### PR TITLE
CAPX-78 | Megans Feedback

### PR DESCRIPTION
**Pull request by sherakama**
# Ready for review
- Removed modules that were for development only.
- Added new help text to various parts of the module
- Updated orphan functionality and new column to capx_profiles table
- Added orphan column view data
- Added orphan column to profiles view
- Added sorting to profile view
- Added adopted functionality to when a profile is added back

### CODE REVIEW

**Comment by sherakama** [comment]
@RoSk0 @mpriest @pingers 

Please review when you have a chance.

### Test
1. Re-install site to get new orphaned field in capx_profiles table
2. Create mapper and importer and importer a few profiles. 
3. Add a sunet id to mapper that is not in the organization
4. Run the import again
5. navigate to /admin/config/capx/profiles and look for sunetid profile
6. Ensure profile is marked as 'No' in orphan column
7. Edit the importer and remove the sunet id
8. Run the 'check for orphans' operation found on /admin/config/capx/importer
9. Navigate back to the profile view /admin/config/capx/profiles
10. Look for the sunetid profile again and ensure it is marked as "Yes" in the orphaned column
11. Go back to the importer and edit it again. Add the sunet that was removed back in.
12. Run the check for orphans operation again
13. Go back to the profile view and look for sunet profile. Ensure that "No" is set in the orphaned column.

File: includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php [hunk 1]
```diff
@@ -362,4 +371,78 @@ public function processOrphans($profileIds) {
 
   }
 
+  /**
+   * Orphaned profiles that have returned to the importer!
+   *
+   * @param  [type] $profiles [description]
+   */
+  public function processAdopted($profiles, $orphans) {
```
Comment (RoSk0) [inline includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php:385]: Document all params with correct descriptions.

File: includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php [hunk 2]
```diff
@@ -362,4 +371,78 @@ public function processOrphans($profileIds) {
...
+        drupal_write_record('capx_profiles', $record, $keys);
+      }
+
+    }
+
+  }
+  /**
+   * Logs that a profile was orphaned.
+   *
+   * @param  Entity $profile
```
Comment (RoSk0) [inline includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php]: "Entity" should be an "object". Becides this is not an entity, it's wrapper.
Take a look at [Doxygen and comment formatting conventions page](http://drupal.org/node/1354).

File: includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php [hunk 3]
```diff
@@ -362,4 +371,78 @@ public function processOrphans($profileIds) {
...
+    );
+
+    $keys = array(
+      'entity_type',
+      'entity_id',
+      'importer',
+      'bundle_type',
+    );
+
+    $yes = drupal_write_record('capx_profiles', $record, $keys);
```
Comment (RoSk0) [inline includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php:448]: Looks like $yes doesn't used anywhere.

File: includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php [hunk 4]
```diff
@@ -362,4 +371,78 @@ public function processOrphans($profileIds) {
...
+
+    $message = t("%title was orphaned from the importer %importername.", array("%title" => $entity->title->value(), "%importername" => $importerName));
```
Comment (RoSk0) [inline includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php]: In `$entity->title->value()` "title" is a node specific property, instead you should use `$entity->label()`.

File: includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php [hunk 5]
```diff
@@ -362,4 +371,78 @@ public function processOrphans($profileIds) {
...
+    watchdog('EntityImporterOrphans', $message, array(), WATCHDOG_NOTICE, '');
```
Comment (RoSk0) [inline includes/CAPx/Drupal/Importer/Orphans/EntityImporterOrphans.php]: You should keep $message transatable, but not translated. Take a look https://api.drupal.org/api/drupal/includes!bootstrap.inc/function/watchdog/7

File: stanford_capx.forms.inc [hunk 6]
```diff
@@ -102,11 +102,11 @@ function stanford_capx_forms_connect_form_validate($form, &$form_state) {
   $endpoint = $form_state['values']['stanford_capx_api_base_url'];
   $authpoint = $form_state['values']['stanford_capx_api_auth_uri'];
 
-  // If password changed validate auth
+  // If password changed validate auth.
   if (!empty($password)) {
     $auth_test = CAPxConnection::testAuthConnection($username, $password, $authpoint);
     if (!$auth_test->status) {
-      form_set_error('stanford_capx_username', 'Could not authenticate with server. Invalid credentials. Please check your username and password');
+      form_set_error('stanford_capx_username', 'Could not authenticate with server. Invalid credentials. Please check your username and password.');
```
Comment (RoSk0) [inline stanford_capx.forms.inc]: Please use t() for user messages.

File: stanford_capx.forms.inc [hunk 7]
```diff
@@ -122,12 +122,12 @@ function stanford_capx_forms_connect_form_validate($form, &$form_state) {
     }
 
     if (!$auth_test->status) {
-      form_set_error("stanford_capx_api_auth_uri", "Please check your authorization url");
+      form_set_error("stanford_capx_api_auth_uri", "Please check your authorization url.");
```
Comment (RoSk0) [inline stanford_capx.forms.inc]: Please use t() for user messages.

**Comment by RoSk0** [comment]
@sherakama Test results:
Followed the instrutions, but on the last step status didn't changed to "No".

**Pull request merged**